src/arch/riscv/isa/templates/vector_mem.isa

```diff
@@ -677,6 +706,7 @@ Fault
     }
     %(op_src_decl)s;
     %(op_rd)s;
+    %(set_vlenb)s;
     %(ea_code)s;
 
     Fault fault = initiateMemRead(xc, traceData, EA, Mem, memAccessFlags);
```

Comment:
ditto

Reply:
I have implemented these changes in a new commit, let me know if I correctly understood what you meant.
Thanks so much!